rust/osm2lanes/src/metric.rs

```diff
@@ -0,0 +1,329 @@
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Metre(f64);
+
+impl Metre {
+    pub const fn new(val: f64) -> Self {
+        Self(val)
+    }
+    pub const fn val(&self) -> f64 {
+        self.0
+    }
+}
+
+impl std::ops::Add for Metre {
+    type Output = Self;
+    fn add(self, other: Self) -> Self {
+        Self(self.0 + other.0)
+    }
+}
+impl std::ops::AddAssign for Metre {
+    fn add_assign(&mut self, other: Self) {
+        *self = Self(self.0 + other.0);
+    }
+}
+impl std::ops::Mul<Metre> for f64 {
+    type Output = Metre;
+    fn mul(self, other: Metre) -> Self::Output {
+        Metre::new(self * other.val())
+    }
+}
+impl std::iter::Sum for Metre {
+    fn sum<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = Metre>,
+    {
+        Self(iter.map(|m| m.0).sum())
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Speed {
```

Comment:
```suggestion

/// Vehicle Speed, used for speed limits and similar
// TODO: This part of the API may need to be revisited entirely
// It is unclear whether a speed unit is needed per lane,
// or if it could be provided for the entire road.
// Data consumers may also want to km/h value
// regardless of the unit used in the highway's locale
// in order to do distance over time calculations.
// The serialization of this is clunky and leaves a lotto be desired.
#[derive(Copy, Clone, Debug, PartialEq)]
pub enum Speed {
```